### PR TITLE
MueLu: More fixes to BrickAggregation for Dirichlet handling

### PR DESCRIPTION
**Pull request by csiefer2**
Tests: RHEL7

And now, with the right target branch!

### CODE REVIEW

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp [hunk 1]
```diff
@@ -211,6 +211,7 @@ namespace MueLu {
       GO aggGID = getAggGID(LID);
       //      printf("[%d] (%d,%d,%d) => agg %d\n",LID,(int)(*xMap_)[x_[LID]],nDim_ > 1 ? (int)(*yMap_)[y_[LID]] : -1,nDim_ > 2 ? (int)(*zMap_)[z_[LID]] : -1,(int)aggGID);
       if(aggGID == GO_INVALID) continue;
+      printf("[%d] getRoot = %d\n",(int)LID,(int)getRoot(LID));
```
Comment (jhux2) [inline packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:214]: This should be guarded, yes?

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp [hunk 2]
```diff
@@ -219,7 +220,7 @@ namespace MueLu {
 
         if (isRoot(LID))
           aggregates->SetIsRoot(LID);
-
+	printf("[%d] initial vertex2AggId = %d\n",(int)LID,(int)vertex2AggId[LID]);
```
Comment (jhux2) [inline packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:223]: ditto above

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp [hunk 3]
```diff
@@ -272,6 +273,11 @@ namespace MueLu {
       }
     }
 
+    for (LO LID = 0; LID < numPoints; LID++) {
+      printf("[%d] vertex2AggId = %d\n",(int)LID,(int)vertex2AggId[LID]);
```
Comment (jhux2) [inline packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:277]: ditto above

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp [hunk 4]
```diff
@@ -400,60 +423,70 @@ namespace MueLu {
...
 
-    /*
+    if(              dirichletX_ && (i == 0 || i == nx_ - 1)) boundary = true;
+    if (nDim_ > 1 && dirichletY_ && (j == 0 || j == ny_ - 1)) boundary = true;
+    if (nDim_ > 2 && dirichletZ_ && (k == 0 || k == nz_ - 1)) boundary = true;
+
+  
     if(boundary)
-      printf("[%d] coord = (%d,%d,%d) {%d,%d} agg = (%d,%d,%d) {%d,%d} => agg %s\n",LID,i,j,k,nx_,ny_,ii,jj,kk,naggx,naggy,"BOUNDARY");
+      printf("[%d] coord = (%d,%d,%d) {%d,%d,%d} agg = (%d,%d,%d) {%d,%d,%d} => agg %s\n",LID,i,j,k,nx_,ny_,nz_,ii,jj,kk,naggx_,naggy_,naggz_,"BOUNDARY");
```
Comment (jhux2) [inline packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:481]: ditto above

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp (hunk 1 above)
Comment (csiefer2) [reply packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:214]: Oops.  Fixed.

File: packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp [hunk 5]
```diff
@@ -400,60 +412,70 @@ namespace MueLu {
...
+    if (nDim_ > 1 && dirichletY_ && (j == 0 || j == ny_ - 1)) boundary = true;
+    if (nDim_ > 2 && dirichletZ_ && (k == 0 || k == nz_ - 1)) boundary = true;
 
     /*
     if(boundary)
-      printf("[%d] coord = (%d,%d,%d) {%d,%d} agg = (%d,%d,%d) {%d,%d} => agg %s\n",LID,i,j,k,nx_,ny_,ii,jj,kk,naggx,naggy,"BOUNDARY");
+      printf("[%d] coord = (%d,%d,%d) {%d,%d,%d} agg = (%d,%d,%d) {%d,%d,%d} => agg %s\n",LID,i,j,k,nx_,ny_,nz_,ii,jj,kk,naggx_,naggy_,naggz_,"BOUNDARY");
     else
-      printf("[%d] coord = (%d,%d,%d) {%d,%d} agg = (%d,%d,%d) {%d,%d} => agg %d\n",LID,i,j,k,nx_,ny_,ii,jj,kk,naggx,naggy,kk*naggy*naggx + jj*naggx + ii);
+      printf("[%d] coord = (%d,%d,%d) {%d,%d,%d} agg = (%d,%d,%d) {%d,%d,%d} => agg %d\n",LID,i,j,k,nx_,ny_,nz_,ii,jj,kk,naggx_,naggy_,naggz_,kk*naggy_*naggx_ + jj*naggx_ + ii);
```
Comment (jhux2) [inline packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:472]: missed one?

Comment (jhux2) [reply packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:472]: nvm

Comment (csiefer2) [reply packages/muelu/src/Graph/BrickAggregation/MueLu_BrickAggregationFactory_def.hpp:472]: Github has led you astray.  Believe in the heart of the code.

**Pull request merged**